api/health/worker.go

```diff
@@ -100,12 +100,12 @@ func (w *worker) RegisterCheck(name string, check Checker, tags ...string) error
 
 	applicationChecks := w.tags[ApplicationTag]
 	isApplicationCheck := applicationChecks.Contains(name)
-
-	w.checks[name] = &taggedChecker{
+	tc := &taggedChecker{
 		checker:            check,
 		isApplicationCheck: isApplicationCheck,
```

Comment:
can move this from above, idc much either way
```suggestion
		isApplicationCheck: applicationChecks.Contains(name),
```